ozark/src/main/java/org/glassfish/ozark/locale/LocaleResolverChain.java

```diff
@@ -0,0 +1,126 @@
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * http://glassfish.java.net/public/CDDL+GPL_1_1.html
+ * or packager/legal/LICENSE.txt.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at packager/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * Oracle designates this particular file as subject to the "Classpath"
+ * exception as provided by Oracle in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package org.glassfish.ozark.locale;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.mvc.locale.LocaleResolver;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Configuration;
+import javax.ws.rs.core.Context;
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Implements the locale resolving algorithm described in {@link LocaleResolver}.
+ *
+ * @author Christian Kaltepoth
+ */
+@ApplicationScoped
+public class LocaleResolverChain {
+
+    @Inject
+    @Any
+    private Instance<LocaleResolver> resolvers;
+
+    @Inject
+    private HttpServletRequest request;
+
+    @Context
+    private Configuration configuration;
+
+    @PostConstruct
+    public void verify() {
+        Objects.requireNonNull(configuration, "The Configuration instance was not injected! " +
+                "Please make sure you are using a recent version of Jersey.");
+    }
+
+    public Locale resolve() {
```

Comment:
Good point! Thanks!

Then we just have to provide whatever information in the context that isn't directly injectable or possible to obtain for the implementers. 

For example, for the current implementation, only the configuration is needed since `HttpServletContext` may be injected as discussed with Christian in the MVC PR.